tests/app/main/helpers/test_frameworks.py

```diff
@@ -758,3 +760,176 @@ def test_only_known_decorated_views_are_protected(self):
+                'Supplier {supplierId} requested "{method} {url}" after {frameworkSlug} applications closed.',
+                extra={
+                    'supplierId': 123,
+                    'method': 'GET',
+                    'url': '/suppliers',
+                    'frameworkSlug': 'g-cloud-9'
+                }
+            )
+        ]
+        assert response[1] == 404
+
+    @pytest.mark.parametrize('status', ('pending', 'standstill', 'live'))
+    @pytest.mark.parametrize('content_set', (True, False))
+    @mock.patch('app.main.helpers.frameworks.content_loader', autospec=True)
+    def test_renders_error_page_correctly_if_following_framework_content_set(
+        self, content_loader_mock, content_set, status
+    ):
+        self.data_api_client_mock.get_framework.return_value = api_stubs.framework(status=status)
+        if content_set:
+            content_loader_mock.get_metadata.return_value = {
+                'name': 'Next Framework 2', 'slug': 'n-f-2', 'coming': '2042'
+            }
+        else:
+            content_loader_mock.get_metadata.side_effect = ContentNotFoundError()
+
+        @return_404_if_applications_closed(lambda: self.data_api_client_mock)
+        def view_function(framework_slug):
+            pass
+
+        with mock.patch('app.main.helpers.frameworks.current_user'):
+            with mock.patch('app.main.helpers.frameworks.current_app'):
+                with self.app.test_request_context('/suppliers'):
+                    response = view_function(framework_slug='g-cloud-9')
+
+        document = html.fromstring(response[0])
+        assert response[1] == 404
+        assert document.xpath('//title/text()')[0] == "Applications closed - Digital Marketplace"
+        assert document.xpath('//h1/text()')[0] == "You can no longer apply to G-Cloud 7"
+        assert "The deadline for applying was 12am GMT, Monday 3 January 2000." in \
+            document.xpath('//div[@class="dmspeak"]/p/text()')[0]
+
+        if content_set:
+            assert "Next Framework 2 is expected to open in 2042." in \
+                document.xpath('//div[@class="dmspeak"]/p/text()')[1]
+        else:
+            assert "is expected to open in" not in response[0]
+
+    def test_returns_the_view_function_if_framework_is_open(self):
+        self.data_api_client_mock.get_framework.return_value = api_stubs.framework(status='open')
+
+        @return_404_if_applications_closed(lambda: self.data_api_client_mock)
+        def view_function(framework_slug):
+            return f'{framework_slug} - OK!'
+
+        response = view_function(framework_slug='g-cloud-7')
+
+        assert response == 'g-cloud-7 - OK!'
+
+    @mock.patch('app.main.views.frameworks.data_api_client')
+    @mock.patch('app.main.views.services.data_api_client')
+    @mock.patch('app.main.helpers.require_login')
+    @mock.patch('app.main.helpers.frameworks.current_user')
+    def test_only_known_decorated_views_are_protected_applicatins(
```

Comment:
Typo, but I do slightly prefer calling them "applicatin's"

Reply:
Ha, I tagged that on the end to differentiate between a different test with the same name and forgot to remote/update it. I'll fix it.